src/schema/document/value.rs

```diff
@@ -146,16 +145,16 @@ pub enum ReferenceValueLeaf<'a> {
     F64(f64),
     /// Date/time with nanoseconds precision
     Date(DateTime),
-    /// Facet
-    Facet(&'a Facet),
+    /// Facet, needs to match the format of [Facet].
+    Facet(&'a str),
     /// Arbitrarily sized byte array
     Bytes(&'a [u8]),
     /// IpV6 Address. Internally there is no IpV4, it needs to be converted to `Ipv6Addr`.
     IpAddr(Ipv6Addr),
     /// Bool value
     Bool(bool),
     /// Pre-tokenized str type,
-    PreTokStr(&'a PreTokenizedString),
+    PreTokStr(Box<PreTokenizedString>),
```

Comment:
We have started using PreTokenizedString in the concatenated field implementation in tantivy.
We use it for bool/numbers.

I am bit scared this PR could involve a non-benine regression for concatenate field.
@trinity-1686a has more info about why it is used.

Reply:
Maybe `Box<Arc<...>>` !?

Reply:
(`Box<Arc<...>>` is just an `Arc<...>` with extra indirection)

Reply:
PreTokenizedString is cloned once already for the tokenization 
`BoxTokenStream::new(PreTokenizedStream::from(tok_str.clone()))`

I think since we can probably remove `PreTokenizedString` from the concat field, we can keep it like this for now and if not revisit it later.

Reply:
@trinity-1686a Usually you would be right but there is a different, and in this very specific case it is a notable one. mem::size of `Box<Arc<..>>` is larger than that of `Arc<...>`.

You might tell me "who cares": overall the earlier in fact takes more memory.
The difference here is that we are in an enum so this options happens to determine the size of all of the num nodes in our tree.

Overall it actually take LESS here to use `Box<Arc<T>>` instead of `Box<T>`.

Anyway @PSeitz moved concatenate to use JSON field instead so we don't care :)

Reply:
I put my foot in my mouth.